Adds `reproject` as an option for `autoalign` plotting of maps (#7772)
sunpy/sunpy: changelog/7772.feature.rst

Added a new option ``autoalign='reproject'`` to :meth:`~sunpy/map/GenericMap/plot`.

> This option allows for automatic reprojection of a map's data to match the WCS of the target axes, using the :meth:`~sunpy/map/GenericMap/reproject_to` method.

Reviewer: This is not how the interlinks work with sphinx unfortunately.

Author: Fixed them

> Added a new option ``autoalign='reproject'`` to :meth:`~sunpy.map.GenericMap.plot`.
> This option allows for automatic reprojection of a map's data to match the WCS of the target axes, using the :meth:`~sunpy.map.GenericMap.reproject_to` method.

Reviewer 2: ```suggestion
Added a new auto-alignment option ``autoalign='reproject'`` to :meth:`~sunpy.map.GenericMap.plot`, which performs a reprojection to the axes WCS.
```
